### ISSUE TITLE
Q: Why ginkgo run specs that belongs to k8s.io/kubernetes/test when I migrate from V1 to V2?

### ISSUE THREAD

**Issue by yuhanqiutg** [onsi/ginkgo#1206]
Hi team, I encountered an issue when I migrate from ginkgo V1 to V2. 
We use ginkgo to write E2E test for our k8s operator. We used ginkgo V1 and it works very well. 
Now we are trying to migrate from ginkgo V1 to V2 and upgrade k8s version at the same time. After we upgraded both ginkgo and kubernetes and run our e2e tests, not only the specs of our project are included, but the specs of k8s are also included. However, when we use V1 with the same code, only specs of our project are included, specs of k8s are excluded.  (When we upgrade from V1 to V2, we didn't change the test code but only replace "github.com/onsi/ginkgo" with "github.com/onsi/ginkgo/v2")

Here is the output when we dry run the tests:
```
[ReportAfterSuite] PASSED [0.026 seconds]
------------------------------

Ran 354 of 358 Specs in 0.025 seconds
SUCCESS! -- 354 Passed | 0 Failed | 0 Pending | 4 Skipped
PASS
```
You can see that there are 358 Specs because specs of k8s are included. 
We could use `focus` to focus on our cases, the output  is:
```
[ReportAfterSuite] PASSED [0.020 seconds]
------------------------------

Ran 116 of 358 Specs in 0.014 seconds
SUCCESS! -- 116 Passed | 0 Failed | 0 Pending | 242 Skipped
PASS
```
That means there are 242 cases are from k8s.io/kubernetes/test but we didn't include them in our code. We just import `"k8s.io/kubernetes/test/e2e/framework"` to help us write test cases for k8s operator. 
We're not sure that whether the issue is caused by upgrading ginkgo or kubernetes.

Our questions are:
1. Does ginkgo change the way to scan specs from V1 to V2 ? (The specs of k8s are not included when we use V1 but included when we migrate to V2, that's strange)
2. How can I exclude these specs from  k8s.io/kubernetes/test ? I use `skip-file` to skip them, but I will see many skips in the test output, and I'm concerned that they will affect performance even if they are skipped because ginkgo have to schedule them to different nodes.

**Comment by onsi** [comment]
hey @yuhanqiutg Ginkgo hasn't fundamentally changed how specs are pulled in between v1 and v2.  Any code that is imported into your compiled test package that calls into Ginkgo's DSL wil generate specs that are added to the suites.  My suspicion is that something somewhere has started importing k8s tests.  I'd recommend looking carefully through your import statements to see if you inadvertently `import "k8s.io.kubernetes/test/...?..."`   If you run with `--dry-run` and `-vv` you should be able to see which files are causing some of these unwanted specs to appear - I'd pick one of them and look for where an import statement might be.

I'm pretty confident that `e2e/framework` should only import framework utilities and not any tests.  But if that isn't a case you should open an issue on kubernetes or ask in the `sig-testing` slack channel.

**Comment by yuhanqiutg** [comment]
Hi @onsi , thanks for your reply and advice! I have dived into the source code in `k8s.io.kubernetes/test/` and finally I found where the specs of k8s are imported.
In our code, we import `"k8s.io/kubernetes/test/e2e/framework/statefulset"`, and this package import `"k8s.io/kubernetes/test/e2e/framework/manifest"`. The package manifest import `"k8s.io/kubernetes/test/e2e/common"` which contains all specs of k8s e2e test. (in branch master and release-1.26 in k8s)
I also looked at the code in branch release-1.22 which we used to include in our project, and I found that in release-1.22 the package manifest doesn't import `"k8s.io/kubernetes/test/e2e/common"`. So I'm sure that this issue is caused by the change of kubernetes project, they import an unexpected package in `e2e/framework`. 
I will close this issue and open another in k8s project.
Thanks again for your reply!
